vital/algo/resection_camera.h

```diff
@@ -0,0 +1,74 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+/**
+ * \file
+ * \brief resection_camera algorithm definition
+ */
+
+#ifndef VITAL_ALGO_RESECTION_CAMERA_H_
+#define VITAL_ALGO_RESECTION_CAMERA_H_
+
+#include <vital/vital_config.h>
+
+#include <vector>
+#include <memory>
+
+#include <vital/algo/algorithm.h>
+#include <vital/types/feature_set.h>
+#include <vital/types/match_set.h>
+#include <vital/types/camera_perspective.h>
+
+namespace kwiver {
+namespace vital {
+namespace algo {
+
+/// An abstract base class to resection a camera using 3D feature
+/// and point projection pairs.
+
+class VITAL_ALGO_EXPORT resection_camera
+  : public kwiver::vital::algorithm_def<resection_camera>
+{
+public:
+  /// Return the name of this algorithm
+  static std::string static_type_name() { return "resection_camera"; }
+
+  /// Estimate camera parameters from 3D points and their corresponding projections
+  ///
+  /// \param [in]  pts2d 2d projections of pts3d in the same order as pts3d
+  /// \param [in]  pts3d 3d points in the same order as pts2d.  Both must be same size.
+  /// \param [in]  init_cal the initial guess intrinsic parameters of the camera
+  /// \param [out] inliers for each point, the value is true if
+  ///                      this pair is an inlier to the estimate
+  virtual
+  kwiver::vital::camera_perspective_sptr
+  resection(std::vector<vector_2d> const& pts2d,
```

Comment:
This is almost identical to the `estimate_pnp` API.  The main difference is that the `init_cal` is an optional input and the output is not guaranteed to use the same intrinsics as the initial guess (if provided).  If this API doesn't change much, we may want to replace `estimate_pnp` with this API and still allow a PNP backend that requires `init_cal` and doesn't modify it.  However, that is something to revisit later.

Reply:
unable to compile with the default arg for init_cal=nullptr and non-default arg for inliers; swapping their order for now, i.e. making init_cal=nullptr the 4th arg, and inliers the 3rd; other ideas are welcome

Reply:
oops, yes sorry.  You can switch the order.  Just also switch the order in the documentation above to match.